tools/crm_resource_runtime.c

```diff
@@ -1288,6 +1288,14 @@ cli_resource_restart(pcmk__output_t *out, pe_resource_t *rsc, const char *host,
     GList *restart_target_active = NULL;
 
     pe_working_set_t *data_set = NULL;
+    pe_resource_t *parent = uber_parent(rsc);
+
+    /* If the implicit resource or primitive resource of a bundle is given, operate on the
+     * bundle itself instead.
+     */
+    if (parent && parent->parent != NULL && parent->parent->variant == pe_container) {
+        rsc = parent->parent;
```

Comment:
FYI there is a convenience function pe_rsc_is_bundled() that could simplify this a bit. Also, uber_parent(rsc) for a primitive is rsc, so uber_parent() is guaranteed to be non-NULL if rsc is.

Reply:
This conditional can be reduced entirely to `if (pe_rsc_is_bundled(rsc))`, which looks much better.